code/model/modelread.cpp

```diff
@@ -1738,14 +1771,37 @@ int read_model_file(polymodel * pm, const char *filename, int n_subsystems, mode
 			}
 
 			case ID_SLDC: // kazan - Shield Collision tree
-				{
+				{   //ShivanSpS - if pof version is 2118 or higher ignore SLDC, otherwise convert it to slc2.
+					if (pm->version < 2118) {
+						mprintf(("SLDC data is being converted to SLC2.\n"));
+						pm->sldc_size = cfread_int(fp);
+						ubyte* sldc_tree, * slc2_tree;
+						sldc_tree = (ubyte*)vm_malloc(pm->sldc_size);
+						slc2_tree = (ubyte*)vm_malloc(pm->sldc_size * 2);
```

Comment:
Please use `std::unique_ptr<ubyte[]>` for automatic deallocation.

Reply:
Ok im new to smart pointers, but you can do that when "cfread(sldc_tree, 1, pm->sldc_size, fp);" expects a void* on sldc_tree?

Reply:
You can still get the "raw" pointer via `.get()` if necessary.